src/box/lua/config/applier/credentials.lua

```diff
@@ -681,13 +681,12 @@ end
 
 local function on_schema_replace_trigger(_, new)
     assert(on_schema_replace_trigger_is_set)
-
-    if new == nil then
+    if new == nil or new[1] ~= 'version' then
```

Comment:
This is the only change in the logic I see within this commit. However, it doesn't look related to the removal of the `mkversion` module.

Can you clarify it in the commit message or separate into another commit?

Reply:
Yep, it's the bug, on replace trigger tried to create version object, when e.g. `replicaset_name` is set

Reply:
Can you add a test?

Reply:
> Can you add a test?

What for? The bug does not exist before replacing `mkversion` with `version`: `mkversion.from_tuple` returned nil, if smth goes wrong, with `version` module used `internal.version_from_tuple` either returns version object or throws an error.  And after replacing `mkversion` with `version` a bunch of tests doesn't work.

Reply:
Seems solved now, thanks!